contracts/votingMachines/GenesisProtocol.sol

```diff
@@ -604,78 +605,52 @@ contract GenesisProtocol is IntVoteInterface {
      * @param _params a parameters array
-     *    _params[0] - _preBoostedVoteRequiredPercentage,
-     *    _params[1] - _preBoostedVotePeriodLimit, //the time limit for a proposal to be in an absolute voting mode.
-     *    _params[2] -_boostedVotePeriodLimit, //the time limit for a proposal to be in an relative voting mode.
-     *    _params[3] -_thresholdConstA
-     *    _params[4] -_thresholdConstB
-     *    _params[5] -_minimumStakingFee
-     *    _params[6] -_quietEndingPeriod
-     *    _params[7] -_proposingRepRewardConstA
-     *    _params[8] -_proposingRepRewardConstB
-     *    _params[9] -_stakerFeeRatioForVoters
-     *    _params[10] -_votersReputationLossRatio
-     *    _params[11] -_votersGainRepRatioFromLostRep
-     *    _params[12] - _daoBountyConst
-     *    _params[13] - _daoBountyLimit
+     *    _params[0] - _quedVoteRequiredPercentage,
+     *    _params[1] - _quedVotePeriodLimit, //the time limit for a proposal to be in an absolute voting mode.
+     *    _params[2] - _boostedVotePeriodLimit, //the time limit for a proposal to be in an relative voting mode.
+     *    _params[3] - _preBoostedVotePeriodLimit, //the time limit for a proposal to be in an preparation state (stable) before boosted.
+     *    _params[4] -_thresholdConstA
+     *    _params[5] -_quietEndingPeriod
+     *    _params[6] -_proposingRepRewardConstA
+     *    _params[7] -_votersReputationLossRatio
+     *    _params[8] -_minimumDaoBounty
+     *    _params[9] -_daoBountyConst
      * @param _voteOnBehalf - authorized to vote on behalf of others.
     */
     function setParameters(
-        uint[14] _params, //use array here due to stack too deep issue.
+        uint[10] _params, //use array here due to stack too deep issue.
         address _voteOnBehalf
     )
     public
     returns(bytes32)
     {
-        require(_params[0] <= 100 && _params[0] > 0,"0 < preBoostedVoteRequiredPercentage <= 100");
-        require(_params[4] > 0 && _params[4] <= 100000000,"0 < thresholdConstB < 100000000 ");
-        require(_params[3] <= 100000000 ether,"thresholdConstA <= 100000000 wei");
-        require(_params[9] <= 100,"stakerFeeRatioForVoters <= 100");
-        require(_params[10] <= 100,"votersReputationLossRatio <= 100");
-        require(_params[11] <= 100,"votersGainRepRatioFromLostRep <= 100");
-        require(_params[2] >= _params[6],"boostedVotePeriodLimit >= quietEndingPeriod");
-        require(_params[7] <= 100000000,"proposingRepRewardConstA <= 100000000");
-        require(_params[8] <= 100000000,"proposingRepRewardConstB <= 100000000");
-        require(_params[12] <= (2 * _params[9]),"daoBountyConst <= 2 * stakerFeeRatioForVoters");
-        require(_params[12] >= _params[9],"daoBountyConst >= stakerFeeRatioForVoters");
+        require(_params[0] <= 100 && _params[0] >= 50,"50 <= quedVoteRequiredPercentage <= 100");
+        require(_params[4] <= 100000000 ether && _params[4] > 1 ether,"1 < thresholdConstA <= 100000000 wei");
```

Comment:
Why use ether denomination? The comment says 100000000 wei?

Reply:
fix at 70e01b9